src/overlays/actors/ovl_En_Shopnuts/z_en_shopnuts.c

```diff
@@ -30,41 +37,265 @@ const ActorInit En_Shopnuts_InitVars = {
+        DECR(this->animFlagAndTimer);
+    }
+    if (!(this->animFlagAndTimer & 0x1000)) {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 0xE38);
+    }
+    if ((this->actor.xzDistFromLink < 120.0f) || (this->animFlagAndTimer == 0x1000)) {
+        EnShopnuts_SetupBurrow(this);
+    } else if (this->animFlagAndTimer == 0) {
+        EnShopnuts_SetupThrowNut(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFA96C.s")
+void EnShopnuts_ThrowNut(EnShopnuts* this, GlobalContext* globalCtx) {
+    Vec3f spawnPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFA9D4.s")
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 0xE38);
+    if (this->actor.xzDistFromLink < 120.0f) {
+        EnShopnuts_SetupBurrow(this);
+    } else if (SkelAnime_Update(&this->skelAnime) != 0) {
+        EnShopnuts_SetupStand(this);
+    } else if (Animation_OnFrame(&this->skelAnime, 6.0f) != 0) {
+        spawnPos.x = this->actor.posRot.pos.x + (Math_SinS(this->actor.shape.rot.y) * 23.0f);
+        spawnPos.y = this->actor.posRot.pos.y + 12.0f;
+        spawnPos.z = this->actor.posRot.pos.z + (Math_CosS(this->actor.shape.rot.y) * 23.0f);
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_NUTSBALL, spawnPos.x, spawnPos.y, spawnPos.z,
+                        this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z, 2) != NULL) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_THROW);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFAA20.s")
+void EnShopnuts_Burrow(EnShopnuts* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
+        EnShopnuts_SetupWait(this);
+    } else {
+        this->collider.dim.height = ((4.0f - CLAMP_MAX(this->skelAnime.curFrame, 4.0f)) * 10.0f) + 5.0f;
+    }
+    if (Animation_OnFrame(&this->skelAnime, 4.0f) != 0) {
+        this->collider.base.acFlags &= ~0x1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFAA7C.s")
+void EnShopnuts_SpawnSalesman(EnShopnuts* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_DNS, this->actor.posRot.pos.x, this->actor.posRot.pos.y,
+                    this->actor.posRot.pos.z, this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z,
+                    this->actor.params);
+        Actor_Kill(&this->actor);
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 0xE38);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFACE0.s")
+void EnShopnuts_ColliderCheck(EnShopnuts* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & 0x2) {
+        this->collider.base.acFlags &= ~0x2;
```

Comment:
```suggestion
    if (this->collider.base.acFlags & 2) {
        this->collider.base.acFlags &= ~2;
```
To make things easier for when collision_check gets updated, don't put the `0x` on these single-digit flags.